libutils/logging.h

```diff
@@ -128,6 +128,23 @@ void LoggingSetColor(bool enabled);
  */
 void LogToSystemLog(const char *msg, LogLevel level);
 
+/**
+ * @brief Log a message with structured data to system log.
+ * @details This function takes a variable-length argument list and expects
+ *          structured data passed as key-value pairs, where both keys and
+ *          values are C-strings. Furthermore, the last pair must contain the
+ *          "MESSAGE" key, and the corresponding value accepts a format-string
```

Comment:
I wish there was some static assert to ensure that there is actually a `MESSAGE` key as the last pair. Because if it's missing, horrible things can easily happen.

Also, if the value has a format sequence and there are no other arguments, it can easily print whatever piece of memory it finds after the list, right?

Please check what the compiler(s) have to offer and there's nothing useful, we need to write a LGTM check.

Reply:
See ticket [CFE-4190](https://northerntech.atlassian.net/browse/CFE-4190)